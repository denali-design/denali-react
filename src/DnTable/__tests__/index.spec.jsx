/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnTable, DnTableDataPositions, DnTableSortDirections } from '../../index'; // src/index.tsx

describe('DnTable Tests', () => {
    it('should export DnTableDataPositions', () => {
        expect.assertions(1);

        expect(DnTableDataPositions).toStrictEqual({
            default: 'default',
            center: 'center',
            left: 'left',
            right: 'right',
        });
    });

    it('should export DnTableSortDirections', () => {
        expect.assertions(1);

        expect(DnTableSortDirections).toStrictEqual({
            default: 'default',
            ascend: 'ascend',
            descend: 'descend',
        });
    });

    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnTable>
                <DnTable.Header>
                    <DnTable.Row>
                        <DnTable.HeaderCell>Date</DnTable.HeaderCell>
                        <DnTable.HeaderCell>Time</DnTable.HeaderCell>
                        <DnTable.HeaderCell>User</DnTable.HeaderCell>
                        <DnTable.HeaderCell>Role</DnTable.HeaderCell>
                        <DnTable.HeaderCell>Cause</DnTable.HeaderCell>
                    </DnTable.Row>
                </DnTable.Header>
                <DnTable.Body>
                    <DnTable.Row>
                        <DnTable.HeaderCell>04/14/2017</DnTable.HeaderCell>
                        <DnTable.Cell>19:34 PDT</DnTable.Cell>
                        <DnTable.Cell>yby.jupiter</DnTable.Cell>
                        <DnTable.Cell>Admin</DnTable.Cell>
                        <DnTable.Cell>jira123</DnTable.Cell>
                    </DnTable.Row>
                </DnTable.Body>
                <DnTable.Footer>
                    <DnTable.Row>
                        <DnTable.Cell>04/14/2017</DnTable.Cell>
                        <DnTable.Cell>19:34 PDT</DnTable.Cell>
                        <DnTable.Cell>yby.jupiter</DnTable.Cell>
                        <DnTable.Cell>Admin</DnTable.Cell>
                        <DnTable.Cell>jira123</DnTable.Cell>
                    </DnTable.Row>
                </DnTable.Footer>
            </DnTable>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnTable isStriped isCards isFrozen>
                <DnTable.Header>
                    <DnTable.Row>
                        <DnTable.HeaderCell position={DnTableDataPositions.right}>Date</DnTable.HeaderCell>
                        <DnTable.HeaderCell isMono>Time</DnTable.HeaderCell>
                        <DnTable.HeaderCell isSorted sortDirection={DnTableSortDirections.ascend}>
                            User
                        </DnTable.HeaderCell>
                        <DnTable.HeaderCell>Role</DnTable.HeaderCell>
                        <DnTable.HeaderCell>Cause</DnTable.HeaderCell>
                    </DnTable.Row>
                </DnTable.Header>
                <DnTable.Body>
                    <DnTable.Row>
                        <DnTable.Cell>04/14/2017</DnTable.Cell>
                        <DnTable.Cell>19:34 PDT</DnTable.Cell>
                        <DnTable.Cell>yby.jupiter</DnTable.Cell>
                        <DnTable.Cell>Admin</DnTable.Cell>
                        <DnTable.Cell>jira123</DnTable.Cell>
                    </DnTable.Row>
                    <DnTable.Row>
                        <DnTable.Cell>04/14/2017</DnTable.Cell>
                        <DnTable.Cell>19:34 PDT</DnTable.Cell>
                        <DnTable.Cell>yby.jupiter</DnTable.Cell>
                        <DnTable.Cell>Admin</DnTable.Cell>
                        <DnTable.Cell>jira123</DnTable.Cell>
                    </DnTable.Row>
                    <DnTable.Row>
                        <DnTable.Cell>04/14/2017</DnTable.Cell>
                        <DnTable.Cell>19:34 PDT</DnTable.Cell>
                        <DnTable.Cell>yby.jupiter</DnTable.Cell>
                        <DnTable.Cell>Admin</DnTable.Cell>
                        <DnTable.Cell>jira123</DnTable.Cell>
                    </DnTable.Row>
                </DnTable.Body>
                <DnTable.Footer>
                    <DnTable.Row>
                        <DnTable.Cell>
                            <a href="#footer">Footer</a>
                        </DnTable.Cell>
                        <DnTable.Cell>
                            <a href="#greylist">pes.acl.greylist</a>
                        </DnTable.Cell>
                        <DnTable.Cell>Foo Turansky</DnTable.Cell>
                        <DnTable.Cell>Admin</DnTable.Cell>
                        <DnTable.Cell>jira123</DnTable.Cell>
                    </DnTable.Row>
                </DnTable.Footer>
            </DnTable>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
