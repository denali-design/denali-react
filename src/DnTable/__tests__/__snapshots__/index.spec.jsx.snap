// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnTable Tests should render correctly with custom props 1`] = `
<div
  class="table-container table-container__freeze"
>
  <table
    class="table is-striped is-cards"
  >
    <thead>
      <tr>
        <th
          class="is-right"
        >
          Date
        </th>
        <th
          class="is-default is-mono"
        >
          Time
        </th>
        <th
          class="is-default is-sorted is-sorted__ascend"
        >
          User
        </th>
        <th
          class="is-default"
        >
          Role
        </th>
        <th
          class="is-default"
        >
          Cause
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="is-default"
        >
          04/14/2017
        </td>
        <td
          class="is-default"
        >
          19:34 PDT
        </td>
        <td
          class="is-default"
        >
          yby.jupiter
        </td>
        <td
          class="is-default"
        >
          Admin
        </td>
        <td
          class="is-default"
        >
          jira123
        </td>
      </tr>
      <tr>
        <td
          class="is-default"
        >
          04/14/2017
        </td>
        <td
          class="is-default"
        >
          19:34 PDT
        </td>
        <td
          class="is-default"
        >
          yby.jupiter
        </td>
        <td
          class="is-default"
        >
          Admin
        </td>
        <td
          class="is-default"
        >
          jira123
        </td>
      </tr>
      <tr>
        <td
          class="is-default"
        >
          04/14/2017
        </td>
        <td
          class="is-default"
        >
          19:34 PDT
        </td>
        <td
          class="is-default"
        >
          yby.jupiter
        </td>
        <td
          class="is-default"
        >
          Admin
        </td>
        <td
          class="is-default"
        >
          jira123
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr>
        <td
          class="is-default"
        >
          <a
            href="#footer"
          >
            Footer
          </a>
        </td>
        <td
          class="is-default"
        >
          <a
            href="#greylist"
          >
            pes.acl.greylist
          </a>
        </td>
        <td
          class="is-default"
        >
          Foo Turansky
        </td>
        <td
          class="is-default"
        >
          Admin
        </td>
        <td
          class="is-default"
        >
          jira123
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`DnTable Tests should render with default props 1`] = `
<table
  class="table"
>
  <thead>
    <tr>
      <th
        class="is-default"
      >
        Date
      </th>
      <th
        class="is-default"
      >
        Time
      </th>
      <th
        class="is-default"
      >
        User
      </th>
      <th
        class="is-default"
      >
        Role
      </th>
      <th
        class="is-default"
      >
        Cause
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th
        class="is-default"
      >
        04/14/2017
      </th>
      <td
        class="is-default"
      >
        19:34 PDT
      </td>
      <td
        class="is-default"
      >
        yby.jupiter
      </td>
      <td
        class="is-default"
      >
        Admin
      </td>
      <td
        class="is-default"
      >
        jira123
      </td>
    </tr>
  </tbody>
  <tfoot>
    <tr>
      <td
        class="is-default"
      >
        04/14/2017
      </td>
      <td
        class="is-default"
      >
        19:34 PDT
      </td>
      <td
        class="is-default"
      >
        yby.jupiter
      </td>
      <td
        class="is-default"
      >
        Admin
      </td>
      <td
        class="is-default"
      >
        jira123
      </td>
    </tr>
  </tfoot>
</table>
`;
